### DISCUSSION TITLE
How to make round cornered progressbar

### DISCUSSION THREAD

**Discussion by sekkit** [AvaloniaUI/Avalonia#10424] [Q&A] [answered]
How to make round cornered progressbar in simpletheme?
the CornerRadius parameter is not working

**Comment by mysteryx93** [comment]
You can copy the [standard styles](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Simple/Controls/ProgressBar.xaml) to create your own and edit as you want.

Or, if you don't want to override everything and just need to set a property, press F12 at runtime to open DevTools, see what control property needs to be altered, you can edit it at runtime, and create a custom style that sets that specific property.

You might need to define a style selector similar to this

    <Style Selector="ProgressBar /template/ ContentPresenter Rectangle">

If it only requires passing a property to an inner control, then Avalonia could consider supporting CornerRadius.

**Comment by sekkit** [comment]
With following code, the progressbar is still not round cornered
```
<ProgressBar x:Name="progressbar" 
	CornerRadius="10" 
	IsIndeterminate="False"
	ShowProgressText="True" 
	Value="{Binding Progress}" 
	MaxHeight="10"  
	Background="LightGray"
	Foreground="#95eb6c">  
	<ProgressBar.Styles>
		<Style Selector="ProgressBar /template/ Border#PART_Indicator">
			<Setter Property="CornerRadius" Value="{Binding CornerRadius}" />
		</Style>
		<Style Selector="ProgressBar /template/ Border#PART_IndeterminateIndicator">
			<Setter Property="CornerRadius" Value="{Binding CornerRadius}" />
		</Style>
	</ProgressBar.Styles> 
</ProgressBar>  
```

**Comment by rabbitism** [comment]
which version？

**Comment by mysteryx93** [comment]
The way you've set your selector, it would look for another ProgressBar within your ProgressBar control.

If it's a style you want for your application, you might want to define that in App.axaml so that you can re-use it. Either for all ProgressBar, or set a style name so that you can use it only when needed.

**Comment by sekkit** [comment]
> 

11-preview -5

**Comment by sekkit** [comment]
> 

I tried to place it to App.xaml but doesn't work

**Comment by timunie** [comment]
How does CornerRadius look like in your ViewModel? Also make sure DataContext is correct. For me the styles works otherwise.

If you have still issues, please share a project with us.

**Comment by timunie** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/47110241/220386254-aede96dc-f353-4465-8e24-3fe8fca42126.png)

as you can see above it should work already. For that reason, let me transfer this issue into a discussion. If you share a simple sample with us, we can have a look what's wrong in your case.

**Reply by sekkit** [reply]
hi, what version are you using?
I'm using preview-5 and I'll try it outside datagrid to see if there is any difference.

**Reply by sekkit** [reply]
I tried and it worked, I found the cause. When I put progressbar inside a datagrid column then it won't rendered as round cornered.

**Reply by timunie** [reply]
Looks like you are binding to it. In DataGrid you have another DataContext (of your item), so make sure the Binding is correctly set / used. Or don't use a Binding but a DynamicResource

**Reply by sekkit** [reply]
ok, thx for your advice

**Reply by timunie** [reply]
marking this as the answer acc. to your reply.
